### ISSUE TITLE
Having trouble connecting to the SDK

### ISSUE THREAD

**Issue by zmanian** [IBM-Blockchain/ibm-blockchain-issues#8]
I have a minimal test case of the SDK.

And I get the following error.
[MemberServicesImpl.enroll] failed to create cert pair: err={"code":14,"metadata":{"_internal_repr":{}}}

```go
var hfc = require('../..');



// Create a client chain.
var chain = hfc.newChain("targetChain");

// Configure the KeyValStore which is used to store sensitive keys
// as so it is important to secure this storage.
chain.setKeyValStore( hfc.newFileKeyValStore('/tmp/keyValStore') );

// Set the URL for member services
chain.setMemberServicesUrl("grpc://*************_ca-discovery.zone.blockchain.ibm.com:30303");

// Add a peer's URL
chain.addPeer("grpc://*********_vp0-discovery.zone.blockchain.ibm.com:30303");

// Enroll "WebAppAdmin" which is already registered because it is
// listed in fabric/membersrvc/membersrvc.yaml with it's one time password.
chain.enroll("WebAppAdmin", "*******", function(err, webAppAdmin) {
   if (err) return console.log("ERROR: failed to register %s: %s",err);
   // Set this user as the chain's registrar which is authorized to register other users.
   chain.setRegistrar(webAppAdmin);
   // Begin listening for web app requests
  //  listenForUserRequests();
});
```

**Comment by dshuffma-ibm** [comment]
So first up I think you should update to the latest HFC which was published last night `0.0.4`. Next I would advise to run in debug mode.

**Comment by zmanian** [comment]
Updating the hfc `0.0.4` had no effect.

I'm zaki at skuchain.com if you would like me to send actual credentials if you want to try and reproduce and urls

I also don't see any sign of the request in the logs from the CA node.

I also checked if the `:30303` was open on the url and the SSL cert was trusted.

Here is the entire DEBUG=* 


  crypto Using sha3-256 +0ms
  hfc Enrolling [req={"enrollmentID":"WebAppAdmin","enrollmentSecret":"*********"}] +60ms
  hfc [MemberServicesImpl.enroll] [{"enrollmentID":"WebAppAdmin","enrollmentSecret":"*********"}] +2ms
  hfc [MemberServicesImpl.enroll] Generating keys... +0ms
  hfc [MemberServicesImpl.enroll] Generating keys...done! +82ms
  hfc [MemberServicesImpl.enroll] Generating request! "3059301306072a8648ce3d020106082a8648ce3d0301070342000454ebd8d92ec61f55fdad355eef9551333a6f89b3e4bf2d0308c8910e486aa65d2f7ef4a6413d5597ec05bf92320613771e29ff8af2b7149e34cb696f3900c057" +3ms
  hfc [MemberServicesImpl.enroll] Adding signing key! +1ms
  hfc [MemberServicesImpl.enroll] Assding encryption key! +0ms
  hfc [MemberServicesImpl.enroll] [Contact ECA] {"ts":{"seconds":{"low":1469807071,"high":0,"unsigned":false},"nanos":0},"id":{"id":"WebAppAdmin"},"tok":{"tok":{"buffer":{"type":"Buffer","data":[51,97,98,51,48,99,57,102,49,52]},"offset":0,"markedOffset":-1,"limit":10,"littleEndian":false,"noAssert":false}},"sign":{"type":0,"key":{"buffer":{"type":"Buffer","data":[48,89,48,19,6,7,42,134,72,206,61,2,1,6,8,42,134,72,206,61,3,1,7,3,66,0,4,84,235,216,217,46,198,31,85,253,173,53,94,239,149,81,51,58,111,137,179,228,191,45,3,8,200,145,14,72,106,166,93,47,126,244,166,65,61,85,151,236,5,191,146,50,6,19,119,30,41,255,138,242,183,20,158,52,203,105,111,57,0,192,87]},"offset":0,"markedOffset":-1,"limit":91,"littleEndian":false,"noAssert":false}},"enc":{"type":0,"key":{"buffer":{"type":"Buffer","data":[48,89,48,19,6,7,42,134,72,206,61,2,1,6,8,42,134,72,206,61,3,1,7,3,66,0,4,224,185,90,63,136,187,252,111,196,71,156,172,23,19,88,128,55,74,40,108,59,133,150,44,189,102,198,5,11,218,85,116,221,122,190,88,80,1,73,130,164,37,229,102,87,61,108,26,152,44,56,5,180,63,150,133,176,195,96,121,112,114,203,81]},"offset":0,"markedOffset":-1,"limit":91,"littleEndian":false,"noAssert":false}},"sig":null}  +0ms
  hfc [MemberServicesImpl.enroll] failed to create cert pair: err={"code":14,"metadata":{"_internal_repr":{}}} +258ms
  hfc [memberServices.enroll] err=Error, enrollment=undefined +1ms
ERROR: failed to register WebAppAdmin: Error

**Comment by smithbk** [comment]
Try adding the following line in **bold**

// Create a client chain.
var chain = hfc.newChain("targetChain");
**chain.setECDSAModeForGRPC(true);**

**Comment by smithbk** [comment]
You'll want to track the issue below which was just opened.  As soon as I get the set of cipher suites to allow, we should be able to make this work out-of-the-box and will do away with the setECDSAModeForGRPC method.

https://github.com/hyperledger/fabric/issues/2349

**Comment by zmanian** [comment]
TLDR doesn't help

Current test code 
```
var hfc = require('hfc');
// Create a client chain.
var chain = hfc.newChain("targetChain");
chain.setECDSAModeForGRPC(true);

// Configure the KeyValStore which is used to store sensitive keys
// as so it is important to secure this storage.
chain.setKeyValStore( hfc.newFileKeyValStore('/tmp/keyValStore') );

// Set the URL for member services
chain.setMemberServicesUrl("grpc://******.zone.blockchain.ibm.com:30303");

// Add a peer's URL
chain.addPeer("grpc://*****.zone.blockchain.ibm.com:30303");

// Enroll "WebAppAdmin" which is already registered because it is
// listed in fabric/membersrvc/membersrvc.yaml with it's one time password.
chain.enroll("WebAppAdmin", "********", function(err, webAppAdmin) {
   if (err) return console.log("ERROR: failed to register WebAppAdmin: %s",err);
   // Set this user as the chain's registrar which is authorized to register other users.
   chain.setRegistrar(webAppAdmin);
   // Begin listening for web app requests
  //  listenForUserRequests();
});
```

Debug output

```
  crypto Using sha3-256 +0ms
  hfc Enrolling [req={"enrollmentID":"WebAppAdmin","enrollmentSecret":"********"}] +68ms
  hfc [MemberServicesImpl.enroll] [{"enrollmentID":"WebAppAdmin","enrollmentSecret":"*******"}] +2ms
  hfc [MemberServicesImpl.enroll] Generating keys... +1ms
  hfc [MemberServicesImpl.enroll] Generating keys...done! +82ms
  hfc [MemberServicesImpl.enroll] Generating request! "3059301306072a8648ce3d020106082a8648ce3d03010703420004c4f1dfed83e0800e3ba9bf80b422164f5330cb2f56f522799d5dfa868ee1807e619cfc877bc9c9167fda23401ac0fcef1df425f19a12d6025bcc0514c0946dfb" +3ms
  hfc [MemberServicesImpl.enroll] Adding signing key! +1ms
  hfc [MemberServicesImpl.enroll] Assding encryption key! +0ms
  hfc [MemberServicesImpl.enroll] [Contact ECA] {"ts":{"seconds":{"low":1470143971,"high":0,"unsigned":false},"nanos":0},"id":{"id":"WebAppAdmin"},"tok":{"tok":{"buffer":{"type":"Buffer","data":[51,97,98,51,48,99,57,102,49,52]},"offset":0,"markedOffset":-1,"limit":10,"littleEndian":false,"noAssert":false}},"sign":{"type":0,"key":{"buffer":{"type":"Buffer","data":[48,89,48,19,6,7,42,134,72,206,61,2,1,6,8,42,134,72,206,61,3,1,7,3,66,0,4,196,241,223,237,131,224,128,14,59,169,191,128,180,34,22,79,83,48,203,47,86,245,34,121,157,93,250,134,142,225,128,126,97,156,252,135,123,201,201,22,127,218,35,64,26,192,252,239,29,244,37,241,154,18,214,2,91,204,5,20,192,148,109,251]},"offset":0,"markedOffset":-1,"limit":91,"littleEndian":false,"noAssert":false}},"enc":{"type":0,"key":{"buffer":{"type":"Buffer","data":[48,89,48,19,6,7,42,134,72,206,61,2,1,6,8,42,134,72,206,61,3,1,7,3,66,0,4,244,247,212,119,36,13,207,218,161,128,6,65,34,252,40,194,143,6,162,26,253,90,217,227,195,201,97,86,182,246,81,187,206,46,242,146,205,233,158,65,173,150,184,162,185,83,25,52,120,84,241,172,206,236,130,98,35,233,51,36,156,167,236,218]},"offset":0,"markedOffset":-1,"limit":91,"littleEndian":false,"noAssert":false}},"sig":null}  +0ms
  hfc [MemberServicesImpl.enroll] failed to create cert pair: err={"code":14,"metadata":{"_internal_repr":{}}} +301ms
  hfc [memberServices.enroll] err=Error, enrollment=undefined +1ms
ERROR: failed to register WebAppAdmin: Error
```

**Comment by zmanian** [comment]
I get `New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES128-GCM-SHA256` as the cipher suites supported by the TLS connection on port.

**Comment by dshuffma-ibm** [comment]
I think we should step back for a second. 

From your original post i see:
```
chain.setMemberServicesUrl("grpc://*************_ca-discovery.zone.blockchain.ibm.com:30303");
```

This won't work as is. You should be using the prototocl `grpcs://` and you should be providing a cert as the 2nd argument to the setMemberServicesUrl() function. I've made a guide to get it working:

Hang tight, I'll adapt my internal guide and post it here in a few minutes.

**Comment by dshuffma-ibm** [comment]
Ok hopefully this gets you in the right direction: [hfc help](https://github.com/IBM-Blockchain/ibm-blockchain-issues/tree/master/hfc_help) @zmanian

**Comment by zmanian** [comment]
I'll give this a try and then we should update https://github.com/zmanian/docs/blob/master/services/blockchain/etn_sdk.md
as the most visible/discoverable doc

**Comment by zmanian** [comment]
@dshuffma-ibm Your guide worked for me except `chain.setECDSAModeForGRPC(false)` is needed to make avoid an error message.

**Comment by dshuffma-ibm** [comment]
hmm interesting, because its the opposite for me. I wonder if high security networks need it as false and the basic ones need it as true... I'll have to look into that.

Well thanks for posting back, glad to know it works for you.

**Comment by shubhamvrkr** [comment]
**i have an issue using nodesdk on bluemix** 

Current test code

var hfc = require('hfc');
// Create a client chain.
var chain = hfc.newChain("targetChain");
chain.setECDSAModeForGRPC(true);

// Configure the KeyValStore which is used to store sensitive keys
// as so it is important to secure this storage.
chain.setKeyValStore( hfc.newFileKeyValStore('/tmp/keyValStore') );

// Set the URL for member services
chain.setMemberServicesUrl("grpc://******.us.blockchain.ibm.com:30303");

// Add a peer's URL
chain.addPeer("grpc://*****.us.blockchain.ibm.com:30303");

// Enroll "WebAppAdmin" which is already registered because it is
// listed in fabric/membersrvc/membersrvc.yaml with it's one time password.
chain.enroll("WebAppAdmin", "********", function(err, webAppAdmin) {
   if (err) return console.log("ERROR: failed to register WebAppAdmin: %s",err);
   // Set this user as the chain's registrar which is authorized to register other users.
   chain.setRegistrar(webAppAdmin);
   // Begin listening for web app requests
  //  listenForUserRequests();
});

i am getting following error:
![capture](https://cloud.githubusercontent.com/assets/17271679/18472927/d1396e20-79d7-11e6-8931-0e7a7952d379.PNG)

How can i solve this?
